### PR TITLE
Add release-1.4 branch configurations for CAPM3/IPAM repos

### PR DESCRIPTION
**Pull request by kashifest**
Updates the prow config and README for CAPM3 and IPAM release-1.4 configurations. The ansible required  tests would be replaced with e2e in future.

### CODE REVIEW

**Review by tuminoid — DISMISSED** [review]

**Comment by kashifest** [comment]
/hold
Until we branch out.

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -156,6 +159,9 @@ branch-protection:
             release-1.3:
               required_status_checks:
                 contexts: ["test-ubuntu-integration-release-1-3"]
+            release-1.4:
+              required_status_checks:
+                contexts: ["test-ubuntu-integration-release-1-4"]
```
[the remark this answers is not in the document]
Comment (kashifest) [reply prow/manifests/overlays/metal3/config.yaml:164]: lets decide that later, for now lets take it as it is.

**Comment by mboukhalfa** [comment]
> _ No description provided. _

Can you please add some description and mention as a todo to replace ansible required tests with the e2e

**Review by tuminoid — APPROVED** [review]
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]
Please configure branch settings in CAPM3/IPAM accordingly after this is merged.

/approve

**Pull request merged**